server/src/com/cloud/tags/TaggedResourceManagerImpl.java

```diff
@@ -204,11 +206,27 @@ public long getResourceId(String resourceId, ResourceObjectType resourceType) {
             accountId = Account.ACCOUNT_ID_SYSTEM;
         }
 
-        if ((domainId == null) || ((accountId != null) && (domainId.longValue() == -1)))
+        if (domainId == null || domainId == -1)
         {
             domainId = _accountDao.getDomainIdForGivenAccountId(accountId);
         }
-        return new Pair<Long, Long>(accountId, domainId);
+        return new Pair<>(accountId, domainId);
+    }
+
+    private void checkResourceAccessible(Long accountId, Long domainId, String exceptionMessage) {
+        Account caller = CallContext.current().getCallingAccount();
+        if ((domainId != null) && (domainId == -1))
```

Comment:
What about using java.util.Objects.equals(Object, Object) here? Then we do not need this `domainId != null`.

Reply:
@rafaelweingartner this looks good. Fixed.